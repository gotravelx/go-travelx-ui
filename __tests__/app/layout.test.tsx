// __tests__/app/layout.test.tsx
import { render, screen } from "@testing-library/react";
import RootLayout from "@/app/layout";

// Mock child components
jest.mock("@/components/main-nav", () => ({
  MainNav: () => <nav data-testid="main-nav">MainNav</nav>,
}));

jest.mock("sonner", () => ({
  Toaster: ({ position }: { position: string }) => (
    <div data-testid="toaster">Toaster {position}</div>
  ),
}));

jest.mock("@/contexts/auth-context", () => ({
  AuthProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="auth-provider">{children}</div>
  ),
}));

jest.mock("@/components/theme-provider", () => ({
  ThemeProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="theme-provider">{children}</div>
  ),
}));

describe("RootLayout", () => {
  it("renders providers and children", () => {
    const { container } = render(
      <RootLayout>
        <div data-testid="page-content">Page Content</div>
      </RootLayout>
    );

    // Check mocked providers/components
    expect(screen.getByTestId("auth-provider")).toBeInTheDocument();
    expect(screen.getByTestId("theme-provider")).toBeInTheDocument();
    expect(screen.getByTestId("main-nav")).toBeInTheDocument();
    expect(screen.getByTestId("toaster")).toBeInTheDocument();

    // Children should be rendered
    expect(screen.getByTestId("page-content")).toBeInTheDocument();

    // (Optional) snapshot to ensure structure
    expect(container).toMatchSnapshot();
  });
});
