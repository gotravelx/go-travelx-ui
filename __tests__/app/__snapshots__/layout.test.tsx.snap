// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RootLayout renders providers and children 1`] = `
<div>
  <html
    lang="en"
  >
    <body
      class="className"
    >
      <div
        data-testid="auth-provider"
      >
        <div
          data-testid="theme-provider"
        >
          <nav
            data-testid="main-nav"
          >
            MainNav
          </nav>
          <div
            data-testid="toaster"
          >
            Toaster 
            top-right
          </div>
          <div
            data-testid="page-content"
          >
            Page Content
          </div>
        </div>
      </div>
    </body>
  </html>
</div>
`;
